coderd/notifications/enqueuer.go

```diff
@@ -80,6 +83,15 @@ func (s *StoreEnqueuer) Enqueue(ctx context.Context, userID, templateID uuid.UUI
 		CreatedBy:              createdBy,
 	})
 	if err != nil {
+		// We have a trigger on the notification_messages table named `inhibit_enqueue_if_disabled` which prevents messages
+		// from being enqueued if the user has disabled them via notification_preferences. The trigger will fail the insertion
+		// with the message "cannot enqueue message: user has disabled this notification".
+		//
+		// This is more efficient than fetching the user's preferences for each enqueue, and centralizes the business logic.
+		if strings.Contains(err.Error(), ErrCannotEnqueueDisabledNotification.Error()) {
```

Comment:
Does `errors.Is` / `errors.Cause` not work for this?

Reply:
What's `errors.Cause`? I don't see that in the stdlib.
`errors.Is` doesn't work because this, strictly speaking, is not the same error.

Reply:
ah Causer is in `pkg/errors`, never mind me